### PR TITLE
Server Tests Part 2: categoryRouter- deleteCategory, getCategoryItems, more cases

### PR DESCRIPTION
**Pull request by Sherry2001**
Based on #21 
Screen Shot 2020-07-14 at 3 57 36 PM

### CODE REVIEW

File: server/api/categoryRouter.js [hunk 1]
```diff
@@ -60,13 +60,18 @@ router.get('/getCategoryItems/:categoryId', async (req, res, next) => {
...
+    if (!categoryData) {
+      throw new Error('CategoryId does not exist in database');
+    }
     response.title = categoryData.title;
     const itemIds = categoryData.items;
 
     const itemObjects = await Item.find({ _id: { $in: itemIds } }).exec();
 
     response.items = itemObjects;
+
```
Comment (Anthony-Fiddes) [inline server/api/categoryRouter.js:74]: nit: perhaps whitespace could be reduced here? I'm not sure it makes the code clearer

File: server/test/categoryRouter.test.js [hunk 2]
```diff
@@ -82,7 +115,6 @@ describe('categoryRouter', () => {
         })
         .set('content-type', 'application/json')
         .end(async function (error, response) {
-          expect(error).to.be.null;
```
Comment (Anthony-Fiddes) [inline server/test/categoryRouter.test.js:85]: Why was this line removed?

File: server/test/categoryRouter.test.js [hunk 3]
```diff
@@ -146,9 +178,46 @@ describe('categoryRouter', () => {
...
+  
+  describe('/getCategoryItems/:categoryId', () => {
+    it('get items in testCategory1, should return response.body.items with 2 items', (done) => {
+      chai
+        .request(server)
+        .get('/category/getCategoryItems/' + testCategory1._id)
+        .end((error, response) => {
+          expect(response).to.have.status(200);
+          expect(response.body.title).equals('User1 Category1');
+          expect(response.body.items.length).equals(2);
```
Comment (Anthony-Fiddes) [inline server/test/categoryRouter.test.js:190]: Use expect('foo').to.have.lengthOf(3) format

File: server/test/categoryRouter.test.js [hunk 4]
```diff
@@ -146,9 +178,46 @@ describe('categoryRouter', () => {
...
   });
+  
+  describe('/getCategoryItems/:categoryId', () => {
+    it('get items in testCategory1, should return response.body.items with 2 items', (done) => {
+      chai
+        .request(server)
+        .get('/category/getCategoryItems/' + testCategory1._id)
+        .end((error, response) => {
+          expect(response).to.have.status(200);
+          expect(response.body.title).equals('User1 Category1');
```
Comment (Anthony-Fiddes) [inline server/test/categoryRouter.test.js:189]: Nit: Maybe change this string to use the title from the variable created above (e.g. category.title or so)? You use the variable for the IDs. Feel free to ignore this if it is bad practice

File: server/test/categoryRouter.test.js (hunk 2 above)
Comment (Sherry2001) [reply server/test/categoryRouter.test.js:85]: Because we never send error bc it's always reflected in the response status, so i figured we wouldn't need

File: server/test/categoryRouter.test.js (hunk 3 above)
Comment (Sherry2001) [reply server/test/categoryRouter.test.js:190]: noted!

File: server/test/categoryRouter.test.js (hunk 4 above)
Comment (Sherry2001) [reply server/test/categoryRouter.test.js:189]: so this one isn't added from the top, so i don't have access to a variable to do .title with. But this will all change bc im re-setting up these tests

**Comment by Sherry2001** [comment]
This PR will be closed. I will resubmit PRs after testing set up changes

File: server/test/categoryRouter.test.js (hunk 2 above)
Comment (Anthony-Fiddes) [reply server/test/categoryRouter.test.js:85]: sgtm
